packages/lib/src/vis/utils.ts

```diff
@@ -342,3 +339,24 @@ export function getCameraFOV(camera: Camera): {
 
   return { topRight, bottomLeft };
 }
+
+export function boundPointToFOV(
+  unboundedPoint: Vector2 | Vector3,
+  camera: Camera
+): Vector2 {
+  const { topRight, bottomLeft } = getCameraFOV(camera);
+  const boundedX = clamp(unboundedPoint.x, bottomLeft.x, topRight.x);
+  const boundedY = clamp(unboundedPoint.y, bottomLeft.y, topRight.y);
+  return new Vector2(boundedX, boundedY);
+}
+
+export function checkModifierKey(
+  modifierKey: ModifierKey | undefined,
+  event: PointerEvent | WheelEvent
+) {
+  if (!modifierKey) {
+    return !event.altKey && !event.ctrlKey && !event.shiftKey;
+  }
+
+  return event.getModifierState(modifierKey);
+}
```

Comment:
Two new utilities to reduce the "cognitive complexity" of `Selection`. `noModifierKeyPressed` was removed as it was used only there.

Not too happy with the naming `checkModifierKey`: if you have a better name...